File: task-03/golang/primeno.go
package main
import "fmt"

func Prime(n int) bool {
	if n <= 2 {
		return true
	} else if n%2 == 0 {
		return false
	}

	for i := 2; i*i <= n; i += 2 {
	    if n%i == 0 {
			return false
		}
	}

	return true
}

func printP(j int) {
	for i := 1; i <= j; i++ {
		if Prime(i) {
			fmt.Println(i)
		}
	}
}
func main() {
	var n int
	fmt.Print("Enter n: ")
	fmt.Scanln(&n)
	printP(n)
}

File: task-03/rust/primeno.rs
use std::io;

fn prime(j: i32) -> bool {
    if j<=1 {
        return false; 
    }
    for i in 2..j {
        if j%i == 0 {
            return false;
        }
    }
    true 
}

fn main() {
    println!("Enter n:");
    
    let mut n = String::new();
    io::stdin()
        .read_line(&mut n)
        .expect("Error");

    let n: i32 = match n.trim().parse() {
        Ok(j) => j,
        Err(_) => {
            println!("Error");
            return;
        }
    };

    for i in 1..=n {
        if prime(i) {
            println!("{}", i);
        }
    }
}

